packages/uploadthing/src/types.ts

```diff
@@ -52,29 +52,29 @@ type ResolverOptions<TParams extends AnyParams> = {
   file: UploadedFile;
 };
 
-type MiddlewareFnArgs<TRuntime> = TRuntime extends "web"
-  ? { req: Request; res: never }
+export type RequestBase = { json: Request["json"]; headers: Headers };
+
+export type RequestAdapter<TRuntime extends string> = 
+  TRuntime extends "pages"
+  ? NextApiRequest & RequestBase
   : TRuntime extends "app"
-  ? { req: NextRequest; res: never }
-  : { req: NextApiRequest; res: NextApiResponse };
+  ? NextRequest
+  : TRuntime extends "web"
+  ? Request
+  : Partial<Request> & RequestBase;
+
+export type ResponseAdapter<TRuntime> = TRuntime extends "pages" ? NextApiResponse : undefined;
 
 type MiddlewareFn<
   TOutput extends Record<string, unknown>,
-  TRuntime extends string,
-> = MiddlewareFnArgs<TRuntime>["res"] extends never
-  ? (req: MiddlewareFnArgs<TRuntime>["req"]) => MaybePromise<TOutput>
+  TRuntime extends string
+> = ResponseAdapter<TRuntime> extends undefined
+  ? (req: RequestAdapter<TRuntime>) => MaybePromise<TOutput>
   : (
-      req: MiddlewareFnArgs<TRuntime>["req"],
-      res: MiddlewareFnArgs<TRuntime>["res"],
+      req: RequestAdapter<TRuntime>,
+      res: ResponseAdapter<TRuntime>
     ) => MaybePromise<TOutput>;
 
-export type ReqMiddlewareFn<TOutput extends Record<string, unknown>> =
-  MiddlewareFn<TOutput, "web">;
-export type NextReqMiddlewareFn<TOutput extends Record<string, unknown>> =
-  MiddlewareFn<TOutput, "app">;
-export type NextApiMiddlewareFn<TOutput extends Record<string, unknown>> =
-  MiddlewareFn<TOutput, "pages">;
```

Comment:
not sure how many are using these but it's technically a breaking change removing these exports

Reply:
it is exported, but it's not used anywhere...

Reply:
Ohh, i got it